backend/wmg/data/load_cube.py

```diff
@@ -13,37 +13,186 @@
+
+    Returns
+    -------
+    snapshot_s3_dest_path: A string that contains the location of the uploaded data
+    """
+    snapshot_id = str(snapshot_id)
+
+    # cleanup files we do not wish to upload
+    shutil.rmtree(f"{snapshot_source_path}/{INTEGRATED_ARRAY_NAME}")
+    shutil.rmtree(f"{snapshot_source_path}/{OBS_ARRAY_NAME}")
+    shutil.rmtree(f"{snapshot_source_path}/{VAR_ARRAY_NAME}")
+
+    snapshot_s3_dest_path = _get_wmg_snapshot_s3_fullpath(
+        snapshot_schema_version, snapshot_id, is_snapshot_validation_successful
+    )
+
+    sync_command = ["aws", "s3", "sync", snapshot_source_path, snapshot_s3_dest_path]
+
+    subprocess.run(sync_command)
+
+    _write_snapshot_metadata(snapshot_schema_version, snapshot_id, is_snapshot_validation_successful)
+
+    return snapshot_s3_dest_path
+
+
+###################################### PRIVATE FUNCTIONS #################################
+
+
+def _get_latest_snapshot_identifier_file_s3_key(snapshot_schema_version: str) -> str:
+    """
+    Return s3 key of latest_snapshot_identifier file
+    """
+    data_root_path = _get_wmg_s3_root_dir_path()
+    _get_wmg_s3_data_schema_dir_path(snapshot_schema_version)
+    file_name = "latest_snapshot_identifier"
+
+    # TODO(prathap) - change to: return f"{data_root_path}/{data_schema_dir_path}/{file_name}"
+    # when working on: https://github.com/chanzuckerberg/single-cell-data-portal/issues/5166
+    return f"{data_root_path}/{file_name}"
+
+
+def _get_latest_snapshot_run_file_s3_key() -> str:
+    """
+    Return s3 key of latest_snapshot_run file
+    """
+    data_root_path = _get_wmg_s3_root_dir_path()
+    file_name = "latest_snapshot_run"
+
+    return f"{data_root_path}/{file_name}"
+
+
+def _get_wmg_s3_bucket_uri():
+    """
+    Get s3 bucket uri
+    """
+    return f"s3://{wmg_bucket_name}"
+
+
+def _get_wmg_s3_root_dir_path():
+    """
+    Get root directory in the s3 bucket under which the data artifact directories will be written to
+    """
+    return stack_name.strip("/") if stack_name else ""
 
-def _get_wmg_bucket_path():
-    if stack_name:
-        return f"s3://{wmg_bucket_name}{stack_name}"
-    else:
-        return f"s3://{wmg_bucket_name}"
+
+def _get_wmg_s3_data_schema_dir_path(snapshot_schema_version: str) -> str:
+    """
+    Get path for a particular data schema version under which data should be located
+    """
+    return f"snapshots/{snapshot_schema_version}"
+
+
+def _get_wmg_snapshot_s3_path(
+    snapshot_schema_version: str, snapshot_id: str, is_snapshot_validation_successful: bool
+) -> str:
+    """
+    Get the path of the directory in which the data artifact is located
+    """
+    snapshot_dir = snapshot_id if is_snapshot_validation_successful else f"{snapshot_id}_validation_failed_snapshot"
+
+    data_root_path = _get_wmg_s3_root_dir_path()
+    _get_wmg_s3_data_schema_dir_path(snapshot_schema_version)
```

Comment:
what is this call suposed to do? i think this needs to be `snapshot_dir = _get_wmg_s3_data_schema_dir_path(snapshot_schema_version)` since we're otherwise not using the return value?

i think it would also be fine to just hard-code `f"snapshots/{snapshot_schema_version}`

Reply:
the left hand variable was removed by the linter. will fix